WooCommerce/Classes/Authentication/Jetpack Setup/LoginJetpackSetupCoordinator.swift

```diff
@@ -33,11 +34,41 @@ final class LoginJetpackSetupCoordinator: Coordinator {
 //
 private extension LoginJetpackSetupCoordinator {
     func showSetupSteps() {
-        let setupUI = LoginJetpackSetupHostingController(siteURL: siteURL, connectionOnly: connectionOnly)
+        let setupUI = LoginJetpackSetupHostingController(siteURL: siteURL, connectionOnly: connectionOnly, onStoreNavigation: { [weak self] in
+            guard let self else { return }
+            self.showStorePickerForLogin()
+        })
         guard let contentNavigationController = navigationController.presentedViewController as? UINavigationController else {
             // this is not likely to happen but handling this for safety
             return navigationController.present(UINavigationController(rootViewController: setupUI), animated: true)
         }
         contentNavigationController.setViewControllers([setupUI], animated: true)
     }
+
+    func showStorePickerForLogin() {
+        storePickerCoordinator = StorePickerCoordinator(navigationController, config: .login)
+
+        // Tries re-syncing to get an updated store list
+        stores.synchronizeEntities { [weak self] in
+            guard let self = self else { return }
+            let matcher = ULAccountMatcher()
+            matcher.refreshStoredSites()
+            guard let matchedSite = matcher.matchedSite(originalURL: self.siteURL) else {
+                DDLogWarn("⚠️ Could not find \(self.siteURL) connected to the account")
+                return
```

Comment:
Super nit: Should we show an error to the user when this happens?

Reply:
Good point! I added an alert in 8cbb669d42978c5aecf4a16dec918c3d3c32ad87 - I'm not sure about showing contact support since we don't have a page for this error in the help center. I'm using the store picker page, but please feel free to suggest another solution.

Reply:
Adding a help centre section for this native Jetpack installation page will be difficult as we are A/B testing this. I am good with your solution. 👍